### PR TITLE
feat: support julia and r & configurable left-side panel supported

### PR DESCRIPTION
**Pull request by sunshinesmilelk**

### CODE REVIEW

File: packages/libro-lab/src/editor-viewer/code-editor-open-handler.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export class CodeEditorViewerOpenHandler extends NavigatableViewOpenHandler<Code
   id = CodeEditorViewerFactory;
 
   canHandle(uri: URI, options?: ViewOpenHandlerOptions) {
-    if (uri.scheme === 'file' && textFileTypes.includes(uri.path.ext)) {
+    if (uri.scheme === 'file' && textFileTypes.includes(uri.path.ext.toLowerCase())) {
```
Comment (BroKun) [inline packages/libro-lab/src/editor-viewer/code-editor-open-handler.ts:17]: Using the `toLowerCase`  method for matching may have scenarios where it is not suitable. I noticed that there is language spec registration information later, which includes an extension list. Is it possible to merge these two? This way, extensions like .ts and .tsx can be compatible.

**Review by huxiamei — APPROVED** [review]

File: packages/libro-lab/src/editor-viewer/module.ts [hunk 2]
```diff
@@ -14,4 +16,6 @@ export const CodeEditorViewerModule = ManaModule.create(
   CodeEditorViewerOpenHandler,
   LibroDefaultViewerOpenHandler,
   JSONLanguageSpecs,
+  RLanguageSpecs,
```
Comment (BroKun) [inline packages/libro-lab/src/editor-viewer/module.ts:19]: Do we have a default way to open text files?

Comment (sunshinesmilelk) [reply packages/libro-lab/src/editor-viewer/module.ts:19]: the part you quoted is used to handle highlighting.

**Comment by BroKun** [comment]
#290

**Pull request merged**